### ISSUE TITLE
amp-analytics unable to track AnalyticsEvents from amp-video-iframes

### ISSUE THREAD

**Issue by Fabio-ff** [ampproject/amphtml#22665]
It does not seem possible to track analytics events triggered by a custom video player integration embedded via an amp-video-iframe.

How to reproduce: 
https://preprod-www.olympicchannel.com/en/stories/news/amp/moustache-record-breaker-lambie-cycling/

In the amp page:
```
<amp-analytics type="segment">

                {
                  "vars": {
                    "writeKey": "*****",
                    "name": "....."
                  },
                  "linkers": {
                    "enabled": "true"
                  },
                  "extraUrlParams": {
                    "type": "PageView",
                    "sentAt": "2019-06-04T07.10.59.909+0",
                    //.......
                  },
                  "triggers": {
                    "click": {
                      "on": "click",
                      "selector": "a",
                      "request": "track",
                      "vars": {
                        "event": "click"
                      },
                      "extraUrlParams": {
                        "type": "Link Clicked",
                        "properties.nonInteraction": "0"
                      }
                    },
              "embed-play": {
              "on": "video-custom-play",
              "request": "track",
              "vars": {
              "event": "Video Content Playing"
              }
              },
              "embed-video-loaded": {
              "on": "video-custom-loaded",
              "request": "track",
              "vars": {
              "event": "Video Loaded"
              }
              }
                  }
                }
            
</amp-analytics>
```

Is supposed to catch and handle both "video-custom-play" and "video-custom-loaded" being triggered from within a <amp-video-iframe> content

as per the following sample:
```
if (!!window.isAmp && !!window.ampIntegration) {
                         window.ampIntegration.postAnalyticsEvent("video-custom-play");
                       }
```

This seems to be affecting all browsers and seems to be there since our very first integration. The only events that the segment implementation of the amp-analytics module are pageview and click on the parent amp page. We could not catch any of the postEvent or postAnlayticsEvent being triggered from within an amp-iframe or an amp-video-iframe.

cc: @calebcordry

**Comment by calebcordry** [comment]
I was able to repro this locally. It seems like whatever events the embed is sending the parent is not listening or routing appropriately, and thus the analytics event is never is never trigged.

to/ @ampproject/wg-analytics

**Comment by zhouyx** [comment]
+ @alanorozco do you think this is an issue on the parent not receiving the message correctly? Thanks

**Comment by aghassemi** [comment]
to @alanorozco to investigate

**Comment by alanorozco** [comment]
Hi @Fabio-ff 

I'm a bit confused about this conditional:

```js
if (!!window.isAmp && !!window.ampIntegration)
```

`window.isAmp` and `window.ampIntegration` are not exposed by the integration library, could I ask where you got that check from?

**Comment by alanorozco** [comment]
@Fabio-ff

Okay, so on one hand this looks like API misusage.

You shouldn't use `window.ampIntegration`. Instead, you should add a callback to the `window.AmpIntegration` array that receives the instance of the class you should push to. Please [look at documentation for the component](https://amp.dev/documentation/components/amp-video-iframe/).


```js
(window.AmpVideoIframe = window.AmpVideoIframe || [])
    .push(function(ampIntegration) {
      // post event from `ampIntegration` instance
      ampIntegration.postAnalyticsEvent('video-custom-loaded');
    });
```

On the other hand, I also found a regression, thanks for reporting this. #24343 is addressing this issue, expect an update here soon.
